src/GitHub.Exports/Services/Logger.cs

```diff
@@ -74,11 +74,19 @@ public static void WriteLine(string message)
         {
             // this codepath is called multiple times on loading
             // doing a little delay so that calling code doesn't get slowed down by this
-            await System.Threading.Tasks.Task.Delay(500);
-            lock(fileLock)
+            try
+            {
+                await System.Threading.Tasks.Task.Delay(500);
+                lock (fileLock)
+                {
+                    File.AppendAllText(defaultLogPath, msg, Encoding.UTF8);
+                }
+            }
+            catch (Exception)
             {
-                File.AppendAllText(defaultLogPath, msg, Encoding.UTF8);
+                throw;
             }
+          
```

Comment:
The `try/catch` should be inside the lock, and only around the `File.AppendAllText`. Also don't throw the exception, if we do that it'll crash Visual Studio.